### DISCUSSION TITLE
Problem of modifying the model 'REGCV2'

### DISCUSSION THREAD

**Discussion by xiaoheimai95** [CURENT/andes#545] [General]
Dear Dr. Cui and Jinning,
I want to modify the `REGCV2` in order to control the voltage. To be more specific, an external control variable `u(t) ` is introduced into the PI controller as an additional input to regulate the reactive power `Q(t)`, which affects the grid voltage. The code I modified in `REGCV2` is as follows:
```
class VSGOuterPIData:
        self.ut = NumParam(default=0.0, tex_name=r'u(t)',
                           info='External control variable for Q(t)',
                           unit='p.u.',
                           power=True,
                           )
class REGCV3ModelBase(Model):
        self.vref2 = Algeb(tex_name=r'v_{ref2}',
                                   info='voltage reference after adjusted by reactive power',
                                   e_str='(u * Qref - Qe + ut) * kv + vref - vref2',
                                   v_str='u * vref')
```
However, it seems no response, is there any other modification I need to do in your opinion?

**Comment by jinningwang** [comment]
Maybe you can plot out the curves of involved variables, then we can know which one didn't work as expected.

**Reply by xiaoheimai95** [reply]
I only added the parameter `ut` and modified `vref2` as mentioned. And the curve of  `vref2` is just the same as the one without modification, which is shown as a straight horizontal line and the value is 1.0.

**Reply by jinningwang** [reply]
Maybe also plot out `Qref`, `Qe`, `kv`, `vref` if they are variables?
